### ISSUE TITLE
scikit-optimize logo

### ISSUE THREAD

**Issue by MechCoder** [scikit-optimize/scikit-optimize#107]
It's time we had a logo. WDYT?

**Comment by glouppe** [comment]
:+1: ideas on what it should represent?

**Comment by glouppe** [comment]
@eamonnmag often has good ideas :)

**Comment by eamonnmag** [comment]
I'll have a think :)

**Comment by MechCoder** [comment]
For starters, I think it should have a snake. On a side note what do the two leaves in sklearn's logo represent (I've often wondered)?

**Comment by raghavrv** [comment]
I can give this a shot. I was thinking of taking the scikit learn ovals and making the gradient descent contours out of it....

**Comment by glouppe** [comment]
> I can give this a shot. I was thinking of taking the scikit learn ovals and making the gradient descent contours out of it....

Thanks! However, not sure gradient descent-like is appropriate for model-based optimisation.

**Comment by raghavrv** [comment]
Ahh okay!!

**Comment by MechCoder** [comment]
@eamonnmag  @raghavrv Thanks for offering to help. Here are a list of Python logo's for inspiration:

https://en.wikipedia.org/wiki/Scikit-image#/media/File:Scikit-image_logo.png
https://en.wikipedia.org/wiki/SymPy#/media/File:Sympy_logo.svg
http://www.pydy.org/
https://en.wikipedia.org/wiki/Scrapy#/media/File:Scrapy_logo.jpg

I am a huge fan of PyDy's and SymPy's logo.
I really like the SymP

**Comment by MechCoder** [comment]
Throwing some ideas around:

1. Simplest: a snake around a black cube (Showing black-box optimization)
2. A snake around scikit-learn's leaves with "learn" replaced with "opt" (Showing that we depend on sklearn's models)
3. A snake around a screwdriver (Showing that we are tuning the parameters)

Any other ideas?

Also pinging @fabianp

**Comment by betatim** [comment]
Yes to logos.

Unsure on 🐍 on a ✈️  eeh logo.

What does an optimal black box look like?

**Comment by betatim** [comment]
Stylised black body radiation spectrum ... 😀

**Comment by MechCoder** [comment]
I also think a snake around a "stylised black box" would be the simplest and most effective :-)

**Comment by betatim** [comment]
@eamonnmag did you have time?

**Comment by betatim** [comment]
v0.1 will not wait for a logo. Code speaks louder than pretty pictures 🚀

**Comment by eamonnmag** [comment]
I dropped the ball on this. Will have a think now :)

**Comment by objectadjective** [comment]
I've attached both png and svg logo concepts.

![scikit-optimize-logo](https://cloud.githubusercontent.com/assets/412257/17908262/8ec92066-6934-11e6-9278-121d879f9467.png)
[scikit-optimize-logo.svg.zip](https://github.com/scikit-optimize/scikit-optimize/files/433481/scikit-optimize-logo.svg.zip)

**Comment by MechCoder** [comment]
@objectadjective Thanks a lot for working on this!

Just curious if you intended the H in the middle to represent hyperparameter?

**Comment by objectadjective** [comment]
@MechCoder I didn't even see that, but I'm just going to say yes :)

**Comment by MechCoder** [comment]
Just two thoughts (ideas) from my side.

1. Would it look better in black?
2, Is it possible to make the space between the H and the O behind as a box (to represent a black box)?

Also curious to know what @betatim , @glouppe and others think..

**Comment by betatim** [comment]
I like it

**Comment by objectadjective** [comment]
Version in black attached. The concept is based on the ouroboros, the snake/python eating itself. It symbolizes infinite self-reflection and introspection. Seemed fitting, eh?

![scikit-optimize-logo-black](https://cloud.githubusercontent.com/assets/412257/18078150/dbaf1ef8-6e3e-11e6-9ec3-bef565df89ae.png)
[scikit-optimize-logo.svg.zip](https://github.com/scikit-optimize/scikit-optimize/files/444130/scikit-optimize-logo.svg.zip)

**Comment by MechCoder** [comment]
Now that you mention it, it makes sense. The most important application is definitely using a machine leaning model to tune the parameters of another machine learning model.

If either one of @betatim or @glouppe give a +1, I'll be happy to add it to the repo and credit you (and whoever that was involved in making this) in AUTHORS.md

**Comment by betatim** [comment]
🐍 👍  for the black version

What other colours/colour combos would be cool? I see t-shirts in our future ... 😀

**Comment by MechCoder** [comment]
How about green for the snake (or the eyes) only? Not sure that will look great.

Do you have any other idea for the colour combination?

**Comment by betatim** [comment]
I was more thinking foreground-background colour combos. I like the logo itself in one colour. But this is really a detail, we should adopt this logo as is already for the github org and then at some point when we have a spare moment we can bike shed over t-shirts :)

By the way: thanks a lot @objectadjective!

**Comment by MechCoder** [comment]
Awesome, I've updated the logo for the github org. I suppose we need it somewhere in the website as well?

@objectadjective Could you have a look at https://github.com/scikit-optimize/scikit-optimize/pull/212 and see if anyone else needs to be added?

**Comment by objectadjective** [comment]
@MechCoder We should add Rose Peng to #212 (I need to find out what her Github handle is). I love the idea of shirts, maybe laptop stickers. We can explore color combinations when we have time, but even the black logo as is would make a nice laptop sticker.

**Comment by MechCoder** [comment]
Done in https://github.com/scikit-optimize/scikit-optimize/commit/d6b6c3c7489f107c8722f5519236a6fac664f9bf . Thanks again!
